File: #119_Pascal's_Triangle_II/Solution.java
import java.util.List;
import java.util.ArrayList;

public class Solution {
    public List<Integer> getRow(int rowIndex) {
    	int index = rowIndex + 1;
        int[] a = new int[index];
        for (int i = 0; i < index; i++) {
	        if (i == 0) a[i] = 1;
	        if (i == 1) a[i] = 1;
	        int prev = a[0];
	        for (int j = 1; j < i; j++) {
	        	int tmp = a[j];
	        	a[j] = a[j] + prev;
	        	prev = tmp;
	        }
            a[i] = 1;
        }
        List<Integer> resList = new ArrayList<Integer>();
        for (int i = 0; i < index; i++) {
        	resList.add(a[i]);
        }
        return resList;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	List<Integer> res = sol.getRow(3);
    	for (Integer it : res) {
    		System.out.print(it + " ");
    	}
    	System.out.println();
    }
}

File: #119_Pascal's_Triangle_II/Solution.py
class Solution:
    # @param {integer} rowIndex
    # @return {integer[]}
    def getRow(self, rowIndex):
        if rowIndex < 0:
        	return []
        if rowIndex == 0:
        	return [1]
        if rowIndex == 1:
        	return [1, 1]
        prev = [1 for x in range(rowIndex + 1)]
        for x in range(2, rowIndex + 1):
        	tmp = 1
        	for y in range(1, x):
        		tmp2 = prev[y]
        		prev[y] = tmp + prev[y]
        		tmp = tmp2
        return prev

sol = Solution()
print sol.getRow(3)